openassessment/xblock/static/js/spec/lms/oa_staff_area.js

```diff
@@ -281,6 +255,137 @@ describe('OpenAssessment.StaffAreaView', function() {
+            });
+
+            it('shows an error message when a cancel submission request fails', function() {
+                // Show the staff area for the test student
+                var staffArea = createStaffArea(),
+                    serverErrorMessage = 'Mock server error';
+                chooseStudent(staffArea, 'testStudent');
+
+                // Cancel the student's submission but return a server error
+                staffArea.comment('Cancellation reason.');
+                server.cancelSubmission = failWith(server, serverErrorMessage);
+                staffArea.cancelSubmission('Bob');
+
+                // Verify that the error message is shown
+                expect($('.cancel-submission-error', staffArea.element).first().text().trim()).toBe(serverErrorMessage);
+            });
+        });
+
+        describe('Staff Grade Override', function() {
+            var fillAssessment = function($assessment) {
+                $('#staff__assessment__rubric__question--2__feedback', $assessment).val('Text response');
+                $('.question__answers', $assessment).each(function() {
+                    $('input[type="radio"]', this).first().click();
+                });
+            };
+
+            var submitAssessment = function(staffArea) {
+                var $assessment = $('.wrapper--staff-assessment', staffArea.element),
+                    $submitButton = $('.action--submit', $assessment);
+                $submitButton.click();
+            };
+
+            it('enables the submit button when all required fields are specified', function() {
+                var staffArea = createStaffArea(),
+                    $assessment, $submitButton;
+                chooseStudent(staffArea, 'testStudent');
+                $assessment = $('.wrapper--staff-assessment', staffArea.element);
+                $submitButton = $('.action--submit', $assessment);
+                expect($submitButton).toHaveClass('is--disabled');
+                fillAssessment($assessment);
+                expect($submitButton).not.toHaveClass('is--disabled');
+            });
+
+            it('can submit a staff grade override', function() {
+                var staffArea = createStaffArea(),
+                    $assessment, $gradeSection;
+                chooseStudent(staffArea, 'testStudent');
+                $assessment = $('.wrapper--staff-assessment', staffArea.element);
+                fillAssessment($assessment);
+
+                // Submit the assessment
+                server.studentTemplate = 'oa_staff_graded_submission.html';
+                submitAssessment(staffArea);
+
+                // Verify that the student info is visible and shows the correct score
+                $gradeSection = $('.staff-info__student__grade', staffArea.element);
+                expect($('.ui-toggle-visibility', $gradeSection)).not.toHaveClass('is--collapsed');
+                expect($('p', $gradeSection).first().text().trim()).toBe(
```

Comment:
Can you also test the score before the override (up around line 357) so we verify that the score was different before?